test/Microsoft.AspNetCore.Routing.Tests/Tree/TreeRouterTest.cs

```diff
@@ -275,6 +274,27 @@ public class TreeRouterTest
         }
 
         [Fact]
+        public async Task TreeRouter_RouteAsync_DoesNotMatchRoutesWithIntermediateDefaultRouteValues()
```

Comment:
Would like to see a few more cases longer/shorter or combinations of default value + optional or default value + catchall